staging/src/k8s.io/kubectl/pkg/generate/versioned/service.go

```diff
@@ -210,6 +211,15 @@ func generateService(genericParams map[string]interface{}) (runtime.Object, erro
 			service.Spec.Ports[i].TargetPort = intstr.FromInt(int(port))
 		}
 	}
+	if len(params["node-port"]) > 0 {
+		nodePort, err := strconv.Atoi(params["node-port"])
+		if err != nil {
+			return nil, fmt.Errorf("invalid node-port: %s", params["node-port"])
+		}
+		for i := range service.Spec.Ports {
+			service.Spec.Ports[i].NodePort = int32(nodePort)
```

Comment:
you can not use the same nodeport for different ports

Reply:
@aojea Thanks! Indeed it fails when 2 port numbers are specified.
```
$ kubectl create deployment test-deploy --image=nginx
$ kubectl expose deployment.apps/test-deploy --type=NodePort --port=80,81 --node-port=30080
The Service "test-deploy" is invalid: spec.ports[1].nodePort: Duplicate value: 30080
```
So, I fixed ```--node-port``` parameter to be able to receive multiple port numbers like ```--port``` parameter.
```
$ kubectl expose deployment.apps/test-deploy --type=NodePort --port=80,81 --node-port=30080,30081
```

And now I found a similar issue on ```--target-port```, I will fix it after this PR.

@eddiezane Sorry, your insight is great. NodePort should be string to receive multiple port numbers.